From f14699792622715c9cec372400f9dc38f1122f63 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenko@redhat.com>
Date: Mon, 25 Jul 2016 11:40:34 +0200
Subject: [PATCH 6/6] ignore STATUS_KEY_CONSIDERED when editing

KEY_CONSIDERED has 2 meanings:
* The key has not been selected
* All subkeys of the key are expired or have been revoked

Both of them are definitely good candidates for ignoring for edit utils.

https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=commit;h=ff71521d9698c7c5df94831a1398e948213af433
https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git;a=commit;h=315fb73d4a774e2c699ac1804f5377559b4d0027

References: https://bugzilla.redhat.com/show_bug.cgi?id=1359521
Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>
---
 gpgme/editutil.py       | 3 ++-
 src/pygpgme-constants.c | 1 +
 tests/test_editkey.py   | 4 +++-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/gpgme/editutil.py b/gpgme/editutil.py
index 617ddc3..267612f 100644
--- a/gpgme/editutil.py
+++ b/gpgme/editutil.py
@@ -60,7 +60,8 @@ def key_editor(function):
                           gpgme.STATUS_KEYEXPIRED,
                           gpgme.STATUS_PROGRESS,
                           gpgme.STATUS_KEY_CREATED,
-                          gpgme.STATUS_ALREADY_SIGNED):
+                          gpgme.STATUS_ALREADY_SIGNED,
+                          gpgme.STATUS_KEY_CONSIDERED):
                 return
             try:
                 data = gen.send((status, args))
diff --git a/src/pygpgme-constants.c b/src/pygpgme-constants.c
index a23b2ef..2435346 100644
--- a/src/pygpgme-constants.c
+++ b/src/pygpgme-constants.c
@@ -141,6 +141,7 @@ static const struct pygpgme_constant constants[] = {
   CONST(STATUS_POLICY_URL),
   CONST(STATUS_BEGIN_STREAM),
   CONST(STATUS_END_STREAM),
+  CONST(STATUS_KEY_CONSIDERED),
   CONST(STATUS_KEY_CREATED),
   CONST(STATUS_USERID_HINT),
   CONST(STATUS_UNEXPECTED),
diff --git a/tests/test_editkey.py b/tests/test_editkey.py
index 4bd2730..d6358f4 100644
--- a/tests/test_editkey.py
+++ b/tests/test_editkey.py
@@ -32,7 +32,9 @@ class EditKeyTestCase(GpgHomeTestCase):
                    'signonly.pub', 'signonly.sec']
 
     def edit_quit_cb(self, status, args, fd):
-        if status in [gpgme.STATUS_EOF, gpgme.STATUS_GOT_IT]:
+        if status in [gpgme.STATUS_EOF,
+                      gpgme.STATUS_GOT_IT,
+                      gpgme.STATUS_KEY_CONSIDERED]:
             return
         self.status = status
         self.args = args
-- 
2.7.4

